tests/functional/aws-node-sdk/test/multipleBackend/put/put.js

```diff
@@ -44,12 +45,12 @@ function awsGetCheck(objectKey, s3MD5, awsMD5, cb) {
         `on call to AWS through S3: ${err}`);
         assert.strictEqual(res.ETag, `"${s3MD5}"`);
         assert.strictEqual(res.Metadata['scal-location-constraint'],
-            awsLocation);
+            location);
         return awsS3.getObject({ Bucket: awsBucket, Key: objectKey },
         (err, res) => {
             assert.strictEqual(err, null, 'Expected success, got error ' +
             `on direct AWS call: ${err}`);
-            if (process.env.ENABLE_KMS_ENCRYPTION === 'true') {
+            if (location === awsLocationEncryption) {
                 // doesn't check ETag because it's different
```

Comment:
please add one more test where you put to the regular aws test location in an encrypted bucket and you check to make sure the data on aws is actually encrypted (so get it directly from aws and check that it is not the same as the data that was first sent to cloudServer).